#+OPTIONS: ^:{}
#+OPTIONS: LaTeX:t
#+TITLE: Using org-mode in research
# #+DATE: \today
#+OPTIONS: toc:nil tags:nil
#+LATEX_CLASS: article
#+LATEX_CLASS_OPTIONS: [a4paper, twoside]
#+LaTeX_HEADER: \usepackage{pdflscape}
#+LaTeX_HEADER: \usepackage{booktabs}
#+LaTeX_HEADER: \usepackage{lscape, subfigure, graphicx}
#+LaTeX_HEADER: \usepackage{minted}
#+LaTeX_HEADER: \usemintedstyle{autumn}
#+LaTeX_HEADER: \newminted{sh}{fontsize=\footnotesize}
#+LATEX_HEADER: \usepackage{hyperref}
#+LATEX_HEADER: \hypersetup{
#+LATEX_HEADER:     colorlinks,%
#+LATEX_HEADER:     citecolor=black,%
#+LATEX_HEADER:     filecolor=black,%
#+LATEX_HEADER:     linkcolor=black,%
#+LATEX_HEADER:     urlcolor=blue}%
#+LATEX_HEADER: \let\tmp\oddsidemargin
#+LATEX_HEADER: \let\oddsidemargin\evensidemargin
#+LATEX_HEADER: \let\evensidemargin\tmp
#+LATEX_HEADER: \reversemarginpar
#+LATEX_HEADER: \makeatletter
#+LATEX_HEADER: \renewcommand{\l@section}{\@dottedtocline{1}{2em}{2em}}
#+LATEX_HEADER: \renewcommand{\l@subsection}{\@dottedtocline{2}{4em}{4em}}
#+LATEX_HEADER: \renewcommand{\l@subsubsection}{\@dottedtocline{3}{6em}{5em}}
#+LATEX_HEADER: \makeatother

* Introduction
  - Emacs is a text editor (short for Editor MAcroS)
  - Text editors are much more useful than to just display/edit text
  - Plain (UTF-8) text is the best form to preserve data (not a Excel
    file not a JPEG image of a plot!)
  - Emacs includes a `mode' called org-mode that is very useful to
    organize data and experiments, carry out analysis using scripts
    and other external programs and store results and all the
    procedures applied to the data in a single location as a simple
    text file.
  - Documenting in org-mode is useful in reproducible research.
* Org-mode in note taking 
** Sections and subsections
   - Section titles have one `*', two `**' indicate subsection titles
   - Easy to increase or decrease levels of different sectioning
   - 
   - an item in a list (see below)
   - 
** Lists
*** Bullet points
   - We already saw how itemized lists are shown in the [[Introduction]]
*** Numbered lists
    - Numbered lists appear like this:
      1. First item
      2. Second item
      3. ...etc.
      4. 
      5. 
      6. 
      7. 
      8. 
*** Check list [3/3]
    - [X] Finish first item
    - [X] Then second item
    - [X] Finally do the third item in the check list
** Todo lists [100%]
*** DONE Download 16S rRNA sequences from NCBI
*** DONE Align to SILVA database using mothur
*** DONE Manually curate using BioEdit, SeaView, etc.
*** DONE Apply hard and soft filter
*** DONE Calculate distance matrix, construct NJ tree
*** DONE Construct ML and MP tree
*** DONE Edit trees in iTOL
*** DONE Beautify in Inkscape or Illustrator

** Tables 
  - Tables can be created by hand
  - Spreadsheet like capablities

|---------+-------|
| Subject | Score |
|---------+-------|
| Phy     |    90 |
| Chem    |    89 |
| Biol    |    70 |
| Math    |    79 |
|---------+-------|
| Total   |   249 |
|---------+-------|
| Avg     |    83 |
|---------+-------|
#+TBLFM: $2=vmean(@2$2..@4$2)
  
   - Tables can also be created from tab-limited or csv-limited text
   - Use command: ~M-x org-table-convert-region~

|---------+-------+-------+-------+-------+---------+------+----------|
| bin     |   N50 |  size |  ctgs | genes | markers | %cov | dup_mark |
|---------+-------+-------+-------+-------+---------+------+----------|
| b1_m4   |  6467 |  7.29 |  1708 |  8588 |     136 |  108 |      294 |
| b2_m4   |  5206 |  5.96 |  1609 |  6954 |      38 |   30 |       23 |
| hi_m4   | 21756 |  1.78 |   123 |  1979 |      58 |   46 |        5 |
| meta_m4 |  4738 | 61.25 | 20730 | 76130 |     139 |  111 |     1407 |
|---------+-------+-------+-------+-------+---------+------+----------|

** Figures
* Org-mode in reproducible research
** Many published results not reproducible
   - A figure or a plot is useful to describe results
   - 53 deliberately chosen cancer research papers (novel approaches)
   - Only 6 were reproducible (11 % cases)
   - 73 % authors: NO RESPONSE to data request (Psychology)
** Solution: Include data with your figures
   - The following illustrate a trivial example
   - But there are real world examples (see email)
** Toy example
     - 454 reads assembled using genome assembly program (Newbler)
     - Two varying parameters:
       - Minimum overlap length in bp(ml): 5, 10, 20, 30, 40, 50
       - Minimum percentage identity (mi): 75, 80, 85, 90, 95
     - The following snippets of code runs newbler gets the stats
*** Newbler assembly
#+BEGIN_SRC sh
# /mnt/hit2g/senthil_files/PYROPHAGE/bin/try_runassembly.sh
#!/bin/bash
# Senthil / UNLV / October 14, 2014
# Try different -ml and -mi values for assembling pyrophage data
for mi in 75 80 85 90 95;
do
    for ml in 5 10 20 30 40 50;
    do
	# with urt
	runAssembly -o ../results/URT_NEW_PYRO_${mi}_${ml} -force \
	    -ml ${ml} -mi ${mi} -nobig -cpu 6 -urt \
	    ../data/Hot_Springs_metagenome_G7162.fasta;
    done
done
#+END_SRC
*** Get assembly stats
   - Use R to check assembly statistics
   - Extract stats to output file
#+BEGIN_SRC sh :results output :file urt.out
  for i in $(find ../results/ -name "454AllContigs.fna");
  do 
      j=$(echo $i | cut -d "/" -f 3);
      k1=$(echo ${j} | cut -d"_" -f 4);
      k2=$(echo ${j} | cut -d "_" -f 5);
      k3=$(echo $j | cut -d "_" -f 1);
      echo -ne "${j}\t${k1}\t${k2}\t\"${k3}\"\t";
      read_fasta -i ${i} | analyze_assembly -x \
          | cut -d ":" -f 2 | tr '\n' '\t' | sed -e 's/---//g';
      echo;
  done > urt.out;
#+END_SRC
*** Experimental output table
#+TBLNAME: urt.out
|--------------------+----+----+------+-------+---------+------|
| name               | mi | ml |  n50 |    lc |   asize | ctgs |
|--------------------+----+----+------+-------+---------+------|
| URT_NEW_PYRO_95_20 | 95 | 20 | 1686 | 17476 | 8327669 | 7880 |
| URT_NEW_PYRO_95_30 | 95 | 30 | 1628 | 17514 | 8246873 | 8012 |
| URT_NEW_PYRO_80_30 | 80 | 30 | 1571 | 15253 | 8427227 | 8594 |
| URT_NEW_PYRO_95_5  | 95 |  5 | 1733 | 17514 | 8294156 | 7524 |
| URT_NEW_PYRO_95_50 | 95 | 50 | 1511 | 17486 | 8058979 | 8446 |
| URT_NEW_PYRO_75_50 | 75 | 50 | 1471 | 16189 | 8256061 | 8971 |
| URT_NEW_PYRO_75_5  | 75 |  5 | 1677 | 17515 | 8490416 | 8100 |
| URT_NEW_PYRO_95_10 | 95 | 10 | 1733 | 17476 | 8290778 | 7563 |
| URT_NEW_PYRO_75_40 | 75 | 40 | 1526 | 15253 | 8340378 | 8746 |
| URT_NEW_PYRO_80_10 | 80 | 10 | 1679 | 17515 | 8499064 | 8069 |
| URT_NEW_PYRO_85_40 | 85 | 40 | 1528 | 15253 | 8314699 | 8702 |
| URT_NEW_PYRO_80_40 | 80 | 40 | 1528 | 11866 | 8333476 | 8752 |
| URT_NEW_PYRO_85_20 | 85 | 20 | 1624 | 15253 | 8535743 | 8465 |
| URT_NEW_PYRO_80_50 | 80 | 50 | 1472 | 15253 | 8246656 | 8947 |
| URT_NEW_PYRO_90_10 | 90 | 10 | 1681 | 17515 | 8494919 | 8079 |
| URT_NEW_PYRO_85_50 | 85 | 50 | 1472 | 15253 | 8245814 | 8930 |
| URT_NEW_PYRO_75_20 | 75 | 20 | 1623 | 15253 | 8533799 | 8483 |
| URT_NEW_PYRO_85_30 | 85 | 30 | 1569 | 15253 | 8432496 | 8618 |
| URT_NEW_PYRO_85_10 | 85 | 10 | 1677 | 17553 | 8499299 | 8089 |
| URT_NEW_PYRO_90_20 | 90 | 20 | 1619 | 15253 | 8528845 | 8487 |
| URT_NEW_PYRO_85_5  | 85 |  5 | 1677 | 17515 | 8492951 | 8077 |
| URT_NEW_PYRO_90_30 | 90 | 30 | 1577 | 15253 | 8423223 | 8583 |
| URT_NEW_PYRO_95_40 | 95 | 40 | 1575 | 17486 | 8125789 | 8085 |
| URT_NEW_PYRO_75_30 | 75 | 30 | 1578 | 14477 | 8420677 | 8547 |
| URT_NEW_PYRO_90_50 | 90 | 50 | 1477 | 15253 | 8231116 | 8889 |
| URT_NEW_PYRO_80_5  | 80 |  5 | 1677 | 13349 | 8496575 | 8092 |
| URT_NEW_PYRO_75_10 | 75 | 10 | 1677 | 17553 | 8498313 | 8074 |
| URT_NEW_PYRO_80_20 | 80 | 20 | 1617 | 16530 | 8547131 | 8506 |
| URT_NEW_PYRO_90_5  | 90 |  5 | 1680 | 17515 | 8490953 | 8065 |
| URT_NEW_PYRO_90_40 | 90 | 40 | 1528 | 15253 | 8330050 | 8703 |
|--------------------+----+----+------+-------+---------+------|
*** Plot 
#+header: 
#+BEGIN_SRC R :results output 
  library(ggplot2)
  mydat <- read.table('urt.out', head=T, row.names=1)
  pdf('n50_ovlgth.pdf', useDingbats=FALSE)
  p <- ggplot(mydat, aes(ml, n50)) +
      geom_point(aes(colour=factor(mi))) +
      xlab('Seq overlap (bp)') +
      ylab('N50 (bp)')
  p <- p + labs(colour='% identity')
  p + ggtitle('Sequence identity and overlap vs N50')
  dev.off()
#+END_SRC

#+RESULTS:
: null device 
:           1 
** Final result
#+CAPTION: N50 is inversely related to sequence identity and overlap
#+LABEL: n50ovl
#+ATTR_LaTeX: width=0.9\textwidth
[[file:./n50_ovlgth.pdf]]

 - The length of the sequence overlap between reads and the N50 are
   inversely related (Fig [[n50ovl]]), higher identity (95%) resulted in
   slightly better N50.

* Exporting to other formats
  - So far, we saw how everything is text (scripts, results,
    documentation, etc)
  - Org-mode allows exporting the text to PDF, HTML
  - Use ~C-e C-l C-p' to get PDF
  - Use ~C-e C-h C-h' to get HTML
